### PR TITLE
Add builder Dockerfile

### PR DESCRIPTION
**Pull request by tbro**
Update docker-compose and friends. Build in CI.

### CODE REVIEW

**Comment by sveitser** [comment]
Could you add it to `scripts/build-docker-images{,-native}` too?

File: scripts/build-docker-images-native [hunk 1]
```diff
@@ -89,4 +89,5 @@ docker build --platform $PLATFORM -t ghcr.io/espressosystems/espresso-sequencer/
 docker build --platform $PLATFORM -t ghcr.io/espressosystems/espresso-sequencer/commitment-task:main -f docker/commitment-task.Dockerfile ${WORKDIR}
 docker build --platform $PLATFORM -t ghcr.io/espressosystems/espresso-sequencer/submit-transactions:main -f docker/submit-transactions.Dockerfile ${WORKDIR}
 docker build --platform $PLATFORM -t ghcr.io/espressosystems/espresso-sequencer/deploy:main -f docker/deploy.Dockerfile ${WORKDIR}
+docker build -t ghcr.io/espressosystems/espresso-sequencer/permissionless-builder:main -f docker/permissionless-builder.Dockerfile ${WORKDIR}
```
Comment (sveitser) [inline scripts/build-docker-images-native:92]: Also needs to be added on line 75.

File: process-compose.yaml [hunk 2]
```diff
@@ -251,6 +251,13 @@ processes:
         path: /healthcheck
       failure_threshold: 100
 
+  permissionless-builder:
+    command: permissionless-builder
+    depends_on:
+      sequencer0:
+        condition: process_healthy
+      da_server:
```
Comment (sveitser) [inline process-compose.yaml:259]: Is there a reason this doesn't have the condition?

**Review by sveitser — APPROVED** [review]
LGTM

File: process-compose.yaml (hunk 2 above)
Comment (tbro) [reply process-compose.yaml:259]: I wasn't sure what it could be, and it wasn't the most urgent issue at this time.

**Comment by sveitser** [comment]
@tbro Does the builder work for you in process compose and the docker demo? It looks like the necessary env vars are not set.

**Pull request merged**